### PR TITLE
Added Bubblesort in Javascript

### LINKED ISSUE #1378: Add Bubble Sort in JavaScript
See requirements: https://sample-programs.therenegadecoder.com/projects/bubble-sort/
Add snippet here: https://github.com/TheRenegadeCoder/sample-programs/tree/master/archive/j/javascript

### PR DESCRIPTION
**Pull request by Cutwell**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [x] I fixed #1378

### Code Snippets

- [x] I named the pull request using `Added/Updated Bubblesort in Javascript` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])

### Documentation

- [ ] I named the pull request using `Added/Updated <Sample Program> in <Language> Article` format
- [ ] I followed the applicable article template
  - [ ] I added/updated a [language article][4] (i.e. The Python Programming Language)
  - [ ] I added/updated a [code snippet article][3] (i.e. Hello World in Perl)
  - [ ] I added/updated a [project article][5] (i.e. Fizz Buzz in Every Language)

### Testing

- [ ] I named the pull request using `Added/Updated <Language>/<Project> Testing` format
- [ ] I followed the testinfo template, if applicable

## Notes

Implemented the Javascript implementation of the Bubblesort algorithm, following the guidelines from issue #1378 and updated the README for the Javascript.

[1]: https://therenegadecoder.com/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[3]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/CODE_ARTICLE_TEMPLATE.md
[4]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/LANGUAGE_ARTICLE_TEMPLATE.md
[5]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/PROJECT_ARTICLE_TEMPLATE.md

### CODE REVIEW

File: archive/j/javascript/bubble-sort.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * Executable function
+ * 
+ * @param {Command line input} input
+ */
+
+function main(input) {
+    // usage text
+    const usage = `Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”`;
```
Comment (jrg94) [inline archive/j/javascript/bubble-sort.js]:
```suggestion
    const usage = `Usage: please provide a list of at least two integers to sort in the format "1, 2, 3, 4, 5"`;
```

**Comment by jrg94** [comment]
Based on some brief testing on my end, I noticed a few issues.

First, no input should produce the error message. In addition, you have the resulting array printing as "[ 1, 2, 3 ]" when we just want to numbers.

Those two things should fix most of the tests.

File: archive/j/javascript/bubble-sort.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        console.log(bubblesort(arr));
+    }
+    else {
+        // invalid input
+        console.log(usage);
+    }
+}
+
+// usage text
+const usage = `Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”`;
```
Comment (jrg94) [inline archive/j/javascript/bubble-sort.js]:
```suggestion
const usage = `Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"`;
```

**Comment by jrg94** [comment]
I'm not sure what errors you're getting now: https://travis-ci.com/TheRenegadeCoder/sample-programs/jobs/246488182

**Comment by Cutwell** [comment]
Issue resolved - was expecting an outputted string with whitespace formatting, but previously the program outputted the array stripped of whitespace.

**Pull request merged**